ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/CommandService.scala

```diff
@@ -262,6 +275,282 @@ final class CommandService(session: LedgerSession) extends LedgerTestSuite(sessi
+          encode(template).getRecord.fields
+        )
+      }
+    }
+
+  private[this] val discloseExerciseToObservers =
+    LedgerTest("CSDiscloseExerciseToObservers", "Diclose exercise to observers") { context =>
+      for {
+        Vector(alpha, beta) <- context.participants(2)
+        Vector(giver, observer1) <- alpha.allocateParties(2)
+        observer2 <- beta.allocateParty()
+        template = WithObservers(giver, Primitive.List(observer1, observer2))
+        withObservers <- alpha.create(giver, template)
+        _ <- alpha.exercise(giver, withObservers.exercisePing)
+        observer1View <- alpha.transactionTrees(observer1)
+        observer2View <- beta.transactionTrees(observer2)
+      } yield {
+        val observer1Exercise = assertSingleton(
+          "The first observer should see exactly one exercise",
+          observer1View.flatMap(exercisedEvents))
+        val observer2Exercise = assertSingleton(
+          "The second observer should see exactly one exercise",
+          observer2View.flatMap(exercisedEvents))
+        assert(
+          observer1Exercise.contractId == observer2Exercise.contractId,
+          "The two observers should see the same exercise")
+        assert(
+          observer1Exercise.contractId == withObservers.unwrap,
+          "The observers shouls see the exercised contract")
+      }
+    }
+
+  private[this] val hugeCommandSubmission =
+    LedgerTest("CSHugeCommandSubmittion", "The server should accept a submission with 15 commands") {
+      context =>
+        {
+          val target = 15
+          for {
+            ledger <- context.participant()
+            party <- ledger.allocateParty()
+            commands = Vector.fill(target)(Dummy(party).create.command)
+            request <- ledger.submitAndWaitRequest(party, commands: _*)
+            _ <- ledger.submitAndWait(request)
+            acs <- ledger.activeContracts(party)
+          } yield {
+            assert(
+              acs.size == target,
+              s"Expected $target contracts to be created, got ${acs.size} instead")
+          }
+        }
+    }
+
+  private[this] val callablePayout =
+    LedgerTest("CSCallablePayout", "Run CallablePayout and return the right events") { context =>
+      for {
+        Vector(alpha, beta) <- context.participants(2)
+        Vector(giver, newReceiver) <- alpha.allocateParties(2)
+        receiver <- beta.allocateParty()
+        callablePayout <- alpha.create(giver, CallablePayout(giver, receiver))
+        tree <- beta.exercise(receiver, callablePayout.exerciseTransfer(_, newReceiver))
+      } yield {
+        val created = assertSingleton("There should only be one creation", createdEvents(tree))
+        assertEquals(
+          "The created event should be the expected one",
+          created.getCreateArguments.fields,
+          encode(CallablePayout(giver, newReceiver)).getRecord.fields)
+      }
+    }
+
+  private[this] val unitAsArgumentToNothing =
```

Comment:
I don't understand what this test is trying to test. Reading the old test doesn't help either. Where does the `Unit` come into play?

Reply:
Good to know I'm not the only clueless person reading this. I'll happily rephrase this, could you suggestions.

Reply:
Kind of good news: this test is actually a duplicate of one found already in the `TransactionService` suite. I'll remove this one, but it's probably worth spending a couple minutes figuring out the meaning of the existing test.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/3320/commits/979de36daf75e8b2382ae37d86ac694511badd63

Reply:
👍